addons/point_of_sale/models/pos_order.py

```diff
@@ -104,7 +104,8 @@ def _process_order(self, order, existing_order):
             # when vals change the state to 'paid'
             for field in ['lines', 'payment_ids']:
                 if order.get(field):
-                    pos_order.write({field: order.get(field)})
+                    existing_records_vals = [r for r in order[field] if r[0] not in [1, 2, 3] or r[1] in pos_order[field].ids]
```

Comment:
```suggestion
                    existing_records_vals = [r for r in order[field] if r[0] not in [1, 2, 3, 4] or r[1] in pos_order[field].ids]
```

I don't know if its possible from PoS but 4 should be taken into account